bottles/backend/managers/steamgriddb.py

```diff
@@ -27,31 +29,42 @@
 
 
 class SteamGridDBManager:
-    @staticmethod
-    def get_game_grid(name: str, config: BottleConfig):
+    def get_steam_game_asset(
+        program_name: str,
+        asset_path: str,
+        asset_type: Optional[str] = None,
+        reraise_exceptions: bool = False,
+    ) -> Optional[str]:
         try:
-            res = requests.get(f"https://steamgrid.usebottles.com/api/search/{name}")
-        except:
-            return
+            # url = f"https://steamgrid.usebottles.com/api/search/{program_name}"
+            url = f"http://127.0.0.1:8000/api/search/{program_name}"
+            if asset_type:
+                url = f"{url}/{asset_type}"
+            res = requests.get(url, timeout=5)
+            res.raise_for_status()
+            filename = SteamGridDBManager.__save_asset_to_steam(res.json(), asset_path)
 
-        if res.status_code == 200:
-            return SteamGridDBManager.__save_grid(res.json(), config)
+        except Exception as e:
+            if isinstance(e, HTTPError):
+                logging.warning(str(e))
+            else:
+                logging.error(str(e))
+            if reraise_exceptions:
+                raise
 
-    @staticmethod
-    def __save_grid(url: str, config: BottleConfig):
-        grids_path = os.path.join(ManagerUtils.get_bottle_path(config), "grids")
-        if not os.path.exists(grids_path):
-            os.makedirs(grids_path)
+        return filename
 
-        ext = url.split(".")[-1]
-        filename = str(uuid.uuid4()) + "." + ext
-        path = os.path.join(grids_path, filename)
+    @staticmethod
+    def __save_asset_to_steam(url: str, asset_path: str) -> str:
+        asset_dir = os.path.dirname(asset_path)
+        if not os.path.exists(asset_dir):
+            os.makedirs(asset_dir)
 
-        try:
-            r = requests.get(url)
-            with open(path, "wb") as f:
-                f.write(r.content)
-        except Exception:
-            return
+        res = requests.get(url)
+        res.raise_for_status()
+        ext = os.path.splitext(url)[-1]
+        asset_path += ext
+        with open(asset_path, "wb") as img:
+            img.write(res.content)
```

Comment:
For increased readability, I wouldn't abbreviate variable names (e.g. `result` as opposed to `res`, etc.)